piper: avoid an ugly traceback

andyholmes opened the pull request:
Occasionally the piper provider will try to pop on an empty work queue. Add simple check to avoid a noisy log.

eeejay commented:
This will make Synthesize fail silently and monopolize the pipe which will put spiel in a bad state. Maybe a new worker should be instantiated instead?

andyholmes commented:
> This will make Synthesize fail silently and monopolize the pipe which will put spiel in a bad state. Maybe a new worker should be instantiated instead?

~~Ah, I see what's happening. Does it make sense to push another 5, or just one?~~

Sorry, now I really see what's happening :)

andyholmes committed:
piper: spawn new workers as necessary

Occasionally the piper provider will try to pop on an empty work queue.
Add simple check and spawn a new worker when necessary.
